### ISSUE TITLE
Zip code Netherlands should allow zipcode without space

### ISSUE THREAD

**Issue by sschurink** [magento/magento2#11898]
**Issue**

Zip code validation warning on adresses from Netherlands.
Current validation is 1234 AA
This should also allow zipcode without space 1234AA

**Preconditions**

On all 2.* versions including development branch.

**Steps to reproduce**

go to cart or checkout, select Netherlands and insert postal code 1234AA

**Expected result**

No warning

**Actual result**

Zipcode gives warning

**Solution** **(completely written out, just copy and paste!)** 

Replace zip countrycode NL with:
```
 <zip countryCode="NL">
      <codes>
        <code id="pattern_1" active="true" example="1234 AB/1234AB">^[0-9]{4}\s?[a-zA-Z]{2}$</code>
      </codes>
    </zip> 
```

in /app/code/Magento/Directory/etc/zip_codes.xml

**Comment by magento-engcom-team** [comment]
@sschurink, thank you for your report.
We've created internal ticket(s) MAGETWO-83058 to track progress on the issue.

**Comment by osrecio** [comment]
Working on it

**Comment by ishakhsuvarov** [comment]
Hi @sschurink 
According to this information from the [post.nl website](https://www.postnl.nl/en/mail-and-parcels/parcels/addressing-tips/) it is recommended to separate postal code digits and letters with a space. 
Do you have any additional information on this?
Thanks

**Comment by sschurink** [comment]
Hai Ievgen,

That is indeed the official way to write it however due to evolving internet forms it is now also an accepted way to add no space between them.
So unofficially both formats are widely accepted in the Netherlands as correct formats.
And we would like to provide that service also without giving a warning box for a correct postal code (in the eyes of almost all Dutch citizens).

Thanks for checking this though!

Van: Ievgen Shakhsuvarov [mailto:notifications@github.com]
Verzonden: vrijdag 3 november 2017 10:56
Aan: magento/magento2 <magento2@noreply.github.com>
CC: Sandra Schurink <sschurink@999games.nl>; Mention <mention@noreply.github.com>
Onderwerp: Re: [magento/magento2] Zip code Netherlands should allow zipcode without space (#11898)


Hi @sschurink<https://github.com/sschurink>
According to this information from the post.nl website<https://www.postnl.nl/en/mail-and-parcels/parcels/addressing-tips/> it is recommended to separate postal code digits and letters with a space.
Do you have any additional information on this?
Thanks

**Comment by ishakhsuvarov** [comment]
@sschurink Thank you for the update.

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-83195

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-83194

**Comment by okorshenko** [comment]
The issue has been fixed in 2.2-develop branch and will be available with 2.2.2 release

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-83196
